phy: titanium/trion rgmii: improvement

maass-hamburg opened the pull request:
improve titanium and trion rgmii phy

maass-hamburg committed:
phy: titaniumrgmii: use multibit io

use multibit io, that was introduced in https://github.com/enjoy-digital/litex/pull/2105

improve and simplify rx and tx

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: trionrgmii: use LiteEthPHYRGMIITX/RX from Titanium

use LiteEthPHYRGMIITX/RX from titanium,
as they are identical.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: titanium/trion rgmii: move ClkOutput

move ClkOutput after the creation
of the PLL.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: titanium/trion rgmii: number ClockDomains

number ClockDomains, so that more than one phy
can be used.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: titaniumrgmii: check clock_pads.rx

If clock_pads.rx is directly connected to a PLL,
connect them directly. These GPIO don't support
beeing used via ClkInput.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg commented:
this https://github.com/enjoy-digital/litex/pull/2163 should be merged before

enjoy-digital commented:
Thanks @maass-hamburg, this looks good. Is it tested on hardware on Trion and Titanium?

maass-hamburg commented:
@enjoy-digital also have to include the latest changes due to the change of the DDRInput of Efinix

maass-hamburg committed:
phy: titaniumrgmii: fix after change of DDRInput now uses pipe mod

fix after change of DDRInput, now that it uses pipeline mode

Also use error  signal that is in rx_ctl.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: titaniumrgmii: use Clocksignals

use Clocksignals, as the Trion expect
these (for DDRInput).

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg committed:
phy: titaniumrgmii: make it work on the ti375 c529 dev board

make it work on the ti375 c529 dev board

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

maass-hamburg commented:
@enjoy-digital this now works on the ti375 c529 dev kit

enjoy-digital commented:
Thanks @maass-hamburg, this looks good then! Nice that the logic duplication is now removed between Trion/Titanium.
